### ISSUE TITLE
sqlite3 dot commands

### ISSUE THREAD

**Issue by Schamschula** [jupyter-xeus/xeus-sqlite#53]
When running `.schema` I get an error:

![Screen Shot 2020-06-18 at 4 30 37 PM](https://user-images.githubusercontent.com/13394058/85073819-16d24d00-b181-11ea-9a05-12f191a55aba.png)

I couldn't find anything in the documentation regarding a workaround.

PS: I've installed xeus-sqlite on macOS using MacPorts. The key issue was getting around libzmq being installed by MacPorts using autotools, rather than cmake. Unfortunately, reinstalling the zmq package using cmake breaks library versions (numbers of current version and compatibility version are lower using cmake than when building with autotools) and would require all other packages depending on it to be rebuilt. So I hacked `cppzmqConfig.cmake` and instances of `link.txt` within xeus-sqlite to get things to correctly link.

`xeus-sqlite.dir/link.txt` was also missing `-lsqlite3` even though cmake correctly found the MacPorts install of sqlite3 (which I maintain).

Currently, the Portfile is in my personal repository:
https://github.com/Schamschula/macports/commit/647ceec1eec99b09e6db9615c846662ff7eae889

**Comment by marimeireles** [comment]
> When running .schema I get an error:

Hi @Schamschula it's because they don't exist.

I'm not super familiar with the SQLite ecosystem and was basing the implementation of xeus-SQLite mostly on [this](https://github.com/SRombauts/SQLiteCpp/) library that's a wrapper around the original SQLite library and my knowledge of the jupyter ecosystem.

From what I could read [here](https://www.w3resource.com/sqlite/sqlite-dot-commands.php#schema) there are a few similarities with what we're doing with Jupyter magics in SQLite and this dot syntax, it's missing a few things though, like `.schema`.

It might be better to change the typical `%` syntax for `.`, what do you think @SylvainCorlay?

Regarding the MacPorts issue, interesting! I have never tried it before. If you want to add an entry on the readmen pointing to your Portfile that'd be much appreciated. :)
Or if you want, I could do it too.

Thanks for the issue @Schamschula.

**Comment by Schamschula** [comment]
Just an update: I figured out the issues with the libraries: I was passing the directory rather than the full path the library itself.

I ended up reinstalling zmq and cppzmq using cmake. I'll see if I can convince the maintainers of those ports to make that switch as well. Now, there no longer are any kludges in the Portfile.

BTW: `OPENSSL_CRYPTO_LIBRARY` should probably be renamed `OPENSSL_LIBRARY` as openssl also provides a `libcrypto`, but we are trying to link against `libssl` instead.
